spec/index.bs

```diff
@@ -3548,6 +3529,74 @@ integration such as source-language debugging.
 
 ### Shader Module Compilation Information ### {#shader-module-compilation-information}
 
+<script type=idl>
+enum GPUCompilationMessageType {
+    "error",
+    "warning",
+    "info"
+};
+
+[Exposed=Window, Serializable]
+interface GPUCompilationMessage {
+    readonly attribute DOMString message;
+    readonly attribute GPUCompilationMessageType type;
+    readonly attribute unsigned long long lineNum;
+    readonly attribute unsigned long long linePos;
+    readonly attribute unsigned long long spanOffset;
+    readonly attribute unsigned long long spanLength;
+};
+
+[Exposed=Window, Serializable]
+interface GPUCompilationInfo {
+    readonly attribute FrozenArray<GPUCompilationMessage> messages;
+};
+</script>
+
+{{GPUCompilationMessage}} has the following attributes:
+
+<dl dfn-type=attribute dfn-for=GPUCompilationMessage>
+    : <dfn>message</dfn>
+    ::
+        A human-readable string containing the message generated during the shader compilation.
+    
+    : <dfn>type</dfn>
+    ::
+        The severity level of the message.
+
+    : <dfn>lineNum</dfn>
+    ::
+        The line number in the shader {{GPUShaderModuleDescriptor/code}} the
+        {{GPUCompilationMessage/message}} corresponds to. Lines are delimeted by the `\n` character.
+        If {{GPUCompilationMessage/message}} corresponds to a span of characters
+        this points to the line one which the span begins.
+
+    : <dfn>linePos</dfn>
+    ::
+        The number of characters, in code units, on line {{GPUCompilationMessage/lineNum}}
```

Comment:
linePos, spanOffset, and spanLength should specify "UTF-16 code units" I think.

Reply:
Done